### PR TITLE
Update the grid, make the orientation switch pretty and add project summary

### PR DESCRIPTION
**Pull request by randomer**

### CODE REVIEW

File: Grid.elm [hunk 1]
```diff
@@ -10,22 +11,15 @@ import Matrix.Extra
...
 import Ship
 
-
 -- Grid
 type alias IsHit = Bool
 type alias Grid = Matrix.Matrix Cell
 type Cell
     = Ship IsHit
     | Empty IsHit
+    -- TODO: Sunk state
```
Comment (john-kelly) [inline Grid.elm]: What do you mean by this? The sunk state is handle by the `Ship IsHit`

Comment (randomer) [inline Grid.elm]: Not quite. There should be also an indication that a ship has been fully destroyed (every cell of a ship was hit)

File: Grid.elm [hunk 2]
```diff
@@ -52,10 +46,52 @@ addShip ship grid =
   shipCoordinates
     |> List.foldr setToShip grid
 
+cellToHtml : Cell -> Html.Html
+cellToHtml cell =
```
Comment (john-kelly) [inline Grid.elm]: So, I'm all for making things look prettier, but not yet. The main focus right now is getting the functionality working. That's a big reason why I am using the `Matrix.prettyPrint` function in the first place. It allows a really nice temp ui while we are creating the game.

File: Main.elm [hunk 3]
```diff
@@ -73,30 +75,69 @@ setupControlsView address player =
...
     , Html.button -- Add Ship
       [ Html.Events.onClick address (SetupAddShip ship.id)
       ]
-      []
+      [ Html.text "Add ship" ]
     ]
   else Html.div [] []
 
+orientationSwitch : Signal.Address Action -> Int -> Ship.Orientation -> Html.Html
+orientationSwitch address shipId orientation =
```
Comment (john-kelly) [inline Main.elm]: So same here. I like the new look, but I think that it will probably get scrapped once we actually start focusing in on the ui. Right now the focus is getting the functionality working. Then we can take a step back and think about how we want to structure the ui -- both from a code standpoint and a interaction standpoint.

Comment (randomer) [inline Main.elm]: Why would it? The thing is that functionality depends on the interface — depending on how you structure your UI, you will need different functions in order to change your model in different ways. That's why I think we should start working on the UI right away — top to bottom approach is more viable I think.

**Comment by john-kelly** [comment]
@randomer Overall. I really like what you did here with the look. So beautiful. I just think that for right now we should get the core functionality working with the computer ai before thinking about ui.

As a side note, I think we should have a brainstorm about where css styles should live. I'm not sure the best way to keep styles maintainable in an elm project.

I am going to leave this PR open so we can reference it in the future.

File: Grid.elm (hunk 1 above)
Comment (john-kelly) [inline Grid.elm]: I see now. This makes sense.

File: Main.elm (hunk 3 above)
Comment (john-kelly) [inline Main.elm]: What I mean is the code for "Switching a ships orientation" is all going to be the same, whether the ui is a radio button, two buttons, or a right mouse click event.

File: Grid.elm (hunk 1 above)
Comment (randomer) [inline Grid.elm]: See? That's a good example why you should focus on the UI first — your imagination won't be enough to think of all the possibilities. Also, while we're on it, there doesn't need to be four fields in the Play view — two will suffice.

File: Main.elm (hunk 3 above)
Comment (john-kelly) [inline Main.elm]: The ui is more of a hook into the functionality. The functionality for changing a ships orientation (and many other functions) is independent from how the ui is implemented. For example, I see the ability to "CLICK" to add ships into the grid as a nice ui feature. That ui feature is going to use the same code for adding ships that the current ui uses.

Comment (randomer) [inline Main.elm]: Not necessarily. For example, you're using four fields to display the game process. However, it's not needed — you need only two. But if you don't think about that, it eventually will affect your model, in which you'll have to take into account that you're using four fields.

Comment (john-kelly) [inline Main.elm]: I'm not sure what you are referring to for this.

Comment (randomer) [inline Main.elm]: This: https://github.com/john-kelly/elm-battleship/blob/master/Player.elm#L104
There needs to be only one grid per player

Comment (john-kelly) [inline Main.elm]: So why is that?

Comment (randomer) [inline Main.elm]: Long story short: http://kortaggio.github.io/battleboat/

Comment (john-kelly) [inline Main.elm]: Okay. So I think that I have though about this one. The design reason for doing it this way is in preparation for networked gameplay. 

The problem with each player only having a single board is that they then need to know the state of their opponents full board in order to draw their tracking board.

This potentially presents a problem with cheating.

Comment (john-kelly) [inline Main.elm]: The opponent could send their board to the server and then the server sends the tracking board to the client, removing the information pertaining to the opponents ship positions. But this is not related to ui at all. This is more of a design decision.

**Comment by randomer** [comment]
Thanks. I agree on the CSS point. There seems to be no way of including CSS when using elm-reactor

**Comment by john-kelly** [comment]
@randomer Is this the most recent code?

**Comment by randomer** [comment]
Yep. The latest version.
